pkg/webui/console/components/webhook-template-form/index.js

```diff
@@ -147,24 +150,29 @@ export default class WebhookTemplateForm extends Component {
     const { name, fields } = webhookTemplate
     const { error, displayOverwriteModal, existingId } = this.state
     const validationSchema = Yup.object({
-      ...fields.reduce(
-        (acc, field) => ({
-          ...acc,
-          [field.id]: Yup.string().required(sharedMessages.validateRequired),
-        }),
-        {
-          webhook_id: Yup.string()
-            .matches(webhookIdRegexp, Yup.passValues(sharedMessages.validateIdFormat))
-            .min(2, Yup.passValues(sharedMessages.validateTooShort))
-            .max(25, Yup.passValues(sharedMessages.validateTooLong))
-            .required(sharedMessages.validateRequired),
-        },
-      ),
+      ...fields
+        .filter(field => !field.optional)
```

Comment:
Not sure we should filter out optional fields. Can still add those to the validation schema but without the `.required` validation? Something like this:

```js
[field.id]: field.optional ? Yup.string() : Yup.string().required(sharedMessages.validateRequired)
```